test/validation/test_formatting_validation.py

```diff
@@ -176,6 +177,21 @@ def test_check_for_fenced_code_block(test_input: str, should_match: bool) -> Non
     assert actual == expected
 
 
+@pytest.mark.parametrize(
+    "test_input,should_match",
+    [
+        ("#This is an unescaped header", True),
+        ("\#This is properly escaped", False),
+        ("# This is meant to render as a header", False)
+    ]
+)
+def test_check_for_unescaped_hashtag(test_input: str, should_match: bool) -> None:
+    """Test if unescaped hashtags are detected."""
+    actual = check_for_unescaped_hashtag(test_input)
+    expected = FormattingIssue.UNESCAPED_HEADER if should_match else None
+    assert actual == expected

```

Comment:
I would make the check a bit "smarter" in order to reduce the number of false positives (see below).

To do that we need to adjust the test cases as well:
- We'd need to add two leading line breaks to every test case
- I think we can add a few more test cases for common patterns. For example:

```
("\n     \n   #heading", True)  # Spaces on the empty line are allowed, up to three spaces before the heading
("\n\n\#hashtag1 #hashtag2", False)  # Only the first hashtag has to be escaped
("\n\n*#hashtag1 #hashtag2*", False)  # Making it italic also formats it as hashtag
```

Reply:
Added the testcases and changed the regex a bit, 9fd4c44